src/mapreduce.jl

```diff
@@ -90,6 +90,36 @@ end
     end
 end
 
+@generated function reduce{D}(op, a::StaticArray, ::Type{Val{D}})
```

Comment:
Oh right... I didn't know about that function. Certainly makes sense to extend them instead.